fitbenchmarking/fitbenchmarking
# Function Evaluation Method

#### Description of Work

Fixes:

1. In every problem object created from either `parse_nist_data.py` or `parse_fitbenchmark_data.py`, there are now two additional modules. 
    1. The first module `eval_f(x, param_list)` is a function evaluation method which takes in x       value(s) and the function parameters, and returns the y value(s) accordingly.
    2. The second module 'get_function()` returns a list of the used function in the fitting process  and the initial parameters

2. For fitting using mantid, the method `mantid.simpleapi.fit` called in `fitbenchmarking/fitting/mantid/main.py` now takes a mantid-native function as a FunctionWrapper object and a user-defined function as a string that describes a function called `fitFunction`. The function `fitFunction` is created from a class `fitFunction` defined in `fitbenchmarking/fitting/mantid/func_def.py`. The class `fitFunction` is a child class of IFunction1D (from mantid.api) and is defined from the function evaluation method of the problem object. All the fitting parameters are declared in the method`init` of the class and the function evaluation method is used in the method `function1D`. 

3. For fitting using scipy, the method `scipy.optimize.curve_fit` called in `fitbenchmarking/fitting/scipy/main.py` now takes the function and the initial parameters from the problem object method, get_function().

#### Testing Instructions

1. Run the example scripts and compare the results to the results produced from the master branch
2. Run the test files to check the correctness of the newly added methods
